### PR TITLE
Updating miscellaneous developer libs

### PR DESCRIPTION
**Pull request by robhudson**
This also adds a few new pre-commit hooks that looked useful, but triggered some file churn, esp `end-of-file-fixer` and `trailing-whitespace`.

I tried keeping the changes in separate commits to better see what file changes were triggered by which changes.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -10,8 +10,15 @@ exclude: >
   )
 repos:
   # Note: hooks that add content must run before ones which check formatting, lint, etc
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+    -   id: check-shebang-scripts-are-executable
+    -   id: check-yaml
+    -   id: end-of-file-fixer
+    -   id: trailing-whitespace
```
Comment (pmac) [inline .pre-commit-config.yaml:19]: We do have markdown files in the repo. We should add `args: [--markdown-linebreak-ext=md]` [like it suggests](https://github.com/pre-commit/pre-commit-hooks#trailing-whitespace) I think.

**Comment by pmac** [comment]
Once this merges we should add some of these commits to the `.git-blame-ignore-revs` file.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (robhudson) [reply .pre-commit-config.yaml:19]: Yeah, good call

Comment (robhudson) [reply .pre-commit-config.yaml:19]: Updated. No `.md` files were harmed in the making of this PR.

**Review by pmac — APPROVED** [review]
🥦

**Pull request merged**